process/block/preprocess/smartContractResults.go

```diff
@@ -496,6 +507,8 @@ func (scr *smartContractResults) ProcessMiniBlock(miniBlock *block.MiniBlock, ha
 			return processedTxHashes, err
 		}
 
+		scr.setInitialBalanceForAddress(miniBlockScrs[index].GetRcvAddr())
```

Comment:
name is confusing. it does not set balance - it initialises the account from the current trie.

Reply:
refactored